exchanges/lbank/lbank_wrapper.go

```diff
@@ -0,0 +1,545 @@
+	var finalResp []exchange.OrderDetail
+	var resp exchange.OrderDetail
+	var tempCurr currency.Pairs
+	var x int
+	if len(getOrdersRequest.Currencies) == 0 {
+		tempCurr = l.GetEnabledCurrencies()
+	} else {
+		for x < len(getOrdersRequest.Currencies) {
+			tempCurr = getOrdersRequest.Currencies
+		}
+	}
+	for a := range tempCurr {
+		p := exchange.FormatExchangeCurrency(l.Name, tempCurr[a])
+		b := int64(1)
+		tempResp, err := l.QueryOrderHistory(p.String(), strconv.FormatInt(b, 10), "200")
+		if err != nil {
+			return finalResp, err
+		}
+		for len(tempResp.Orders) != 0 {
+			tempResp, err = l.QueryOrderHistory(p.String(), strconv.FormatInt(b, 10), "200")
+			if err != nil {
+				return finalResp, err
+			}
+			for x := 0; x < len(tempResp.Orders); x++ {
+				resp.Exchange = l.GetName()
+				resp.CurrencyPair = currency.NewPairFromString(tempResp.Orders[x].Symbol)
+				if strings.EqualFold(tempResp.Orders[x].Type, "buy") {
+					resp.OrderSide = exchange.BuyOrderSide
+				} else {
+					resp.OrderSide = exchange.SellOrderSide
+				}
+				z := tempResp.Orders[x].Status
+				switch {
+				case z == -1:
+					resp.Status = "cancelled"
+				case z == 1:
+					resp.Status = "on trading"
+				case z == 2:
+					resp.Status = "filled partially"
+				case z == 3:
+					resp.Status = "Filled totally"
+				case z == 4:
+					resp.Status = "Cancelling"
+				default:
+					resp.Status = "Invalid Order Status"
+				}
+				resp.Price = tempResp.Orders[x].Price
+				resp.Amount = tempResp.Orders[x].Amount
+				resp.OrderDate = time.Unix(tempResp.Orders[x].CreateTime, 9)
+				resp.ExecutedAmount = tempResp.Orders[x].DealAmount
+				resp.RemainingAmount = tempResp.Orders[x].Price - tempResp.Orders[x].DealAmount
+				resp.Fee, err = l.GetFeeByType(&exchange.FeeBuilder{
+					FeeType:       exchange.CryptocurrencyTradeFee,
+					Amount:        tempResp.Orders[x].Amount,
+					PurchasePrice: tempResp.Orders[x].Price})
+				if err != nil {
+					resp.Fee = lbankFeeNotFound
+				}
+				finalResp = append(finalResp, resp)
+				b++
+			}
+		}
+	}
+	return finalResp, nil
+}
+
+// GetFeeByType returns an estimate of fee based on the type of transaction *
+func (l *Lbank) GetFeeByType(feeBuilder *exchange.FeeBuilder) (float64, error) {
+	var resp float64
+	if feeBuilder.FeeType == exchange.CryptocurrencyTradeFee {
+		return feeBuilder.Amount * feeBuilder.PurchasePrice * l.Fee, nil
+	}
+	if feeBuilder.FeeType == exchange.CryptocurrencyWithdrawalFee {
+		withdrawalFee, err := l.GetWithdrawConfig(feeBuilder.Pair.Base.Lower().String())
+		if err != nil {
+			return resp, err
+		}
+		var tempFee string
+		temp := strings.Split(withdrawalFee[0].Fee, ":\"")
+		if len(temp) > 1 {
+			tempFee = strings.TrimRight(temp[1], ",\"type")
+		} else {
+			tempFee = temp[0]
+		}
+		resp, err = strconv.ParseFloat(tempFee, 64)
+		if err != nil {
+			return resp, err
+		}
+	}
+	return resp, nil
+}
+
+// GetAllOpenOrderID returns map[string][]string -> map[currencypair][]orderIDs
```

Comment:
This comment is a little strange since its referencing actual code. I think a better one would be `GetAllOpenOrderID returns all open orders by currency pairs` but I don't think this function should be exported anyway as its not part of IBotExchange and is a helper function